controllers/scopeinstance_controller.go

```diff
@@ -374,3 +457,14 @@ func (r *ScopeInstanceReconciler) mapToScopeInstance(obj client.Object) (request
 
 	return
 }
+
+func (r *ScopeInstanceReconciler) updateScopeInstanceCondition(ctx context.Context, si *operatorsv1.ScopeInstance, condition metav1.Condition) error {
+	// Update the condition of the ScopeInstance
+	meta.SetStatusCondition(&si.Status.Conditions, condition)
+	si.ManagedFields = nil
+	return r.Status().Patch(ctx,
+		si,
+		client.Apply,
+		client.FieldOwner(siCtrlFieldOwner),
+		client.ForceOwnership)
```

Comment:
Using a concrete API object with server-side apply patching is not recommended because any non-pointer fields in the struct will be serialized as the default value for the type and sent as part of the patch even if you didn't explicitly set it. See: https://github.com/operator-framework/rukpak/blob/c9aec01cfe3ec8cfe2d59d2802096ff7ee1d5ffe/internal/rukpakctl/run.go#L115-L119

The better alternatives are to use unstructured.Unstructured or to generate and use applyconfiguration structs for your APIs.

In this specific case, since you're mutating what you just fetched from your cache, it should be fine to use `Update` instead of `Patch`.

Reply:
> Using a concrete API object with server-side apply patching is not recommended because any non-pointer fields in the struct will be serialized as the default value for the type and sent as part of the patch even if you didn't explicitly set it. See: https://github.com/operator-framework/rukpak/blob/c9aec01cfe3ec8cfe2d59d2802096ff7ee1d5ffe/internal/rukpakctl/run.go#L115-L119
>
> The better alternatives are to use unstructured.Unstructured or to generate and use applyconfiguration structs for your APIs.

Ah okay. That makes sense, thanks for pointing that out!

> In this specific case, since you're mutating what you just fetched from your cache, it should be fine to use Update instead of Patch.

The reason I went with `Patch` was because of the implementation of using server-side apply for all of our updates in #24 .
Now that you have mentioned the use applyconfigurations or unstructured.Unstructured I think we may need to update the `Patch` calls we make to use one of these.

Reply:
In general, SSA is preferred when creating/managing objects that you expect another controller to reconcile. For objects that your controller is reconciling, you're typically only ever touching the status, and there's generally an expectation that only a single controller writes to the status.

So my advice would be:
1. For objects you're reconciling: use status update
2. For objects you're creating/managing: use SSA

And only deviate from that model if you have a specific reason to.

Reply:
That makes sense. Thanks for clearing that up for me!